### PR TITLE
Add fancy placeholders for Marketing components

### PR DESCRIPTION
**Pull request by tomalec**
Add placeholders for 

- [x] RecommendedExtensionsItem.
- [x] KnowledgeBase.

Addresses https://github.com/woocommerce/woocommerce-admin/issues/5375.

### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [x] I've tested using a screen reader
- [x] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots
GIF animation of placeholders
https://d.pr/i/uTkAtS

### Detailed test instructions:

1. Ex: Open page `http://wptest.local/wordpress/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`
2. Scroll down if needed,
3. If the content blinks too fast you can throttle network speed in dev tools.
4. Observe the "Recommended extensions" and "Knowledge base" sections, before the content is fully loaded.

### CODE REVIEW

**Comment by tomalec** [comment]
I'm still not sure about file names here. At first, I tried to adapt to the way the existing `placeholder.js` was created. On the other hand, it resulted in more `placeholder.js` files.

Personally, I don't like having different files with different classes called the same way. It makes file searching harder <kbd>Ctrl</kbd> + <kbd>p</kbd>. In some previous projects, I used to work with conventions to call the file like the class it defines if it defines only one, or at least give it a somewhat-specific module name.

**Review by jconroy — COMMENTED** [review]
Niiiice @tomalec 👍 

Testing well for me with the exception of the icons for the coupon page recommended extensions - the border is still being shown for the placeholders e.g. https://d.pr/v/Nv0QEA

Thanks for adding/updating tests 🙇

**Comment by jconroy** [comment]
> [I'd appreciate some comment on that.](https://github.com/woocommerce/woocommerce-admin/pull/5611#issuecomment-726428767)

Honestly not sure if wc-admin has a convention in place. As you've found there are already a few other uses of `placeholder.js` so I think for now it'd be safe to just take the same approach.

I definitely see your point though - not sure if @timmyc has any guideance for us.

File: client/marketing/components/knowledge-base/placeholder.js
Comment (jconroy) [inline client/marketing/components/knowledge-base/placeholder.js:17]: JS tests are flagging a couple of errors for the headings `17:5  error  Headings must have content and the content must be accessible by a screen reader  jsx-a11y/heading-has-content`

Wonder if we can ignore or use another element like a div here? I think the inbox panel might use a div instead of a heading for this

File: client/marketing/components/recommended-extensions/placeholder.js
Comment (ecgan) [inline client/marketing/components/recommended-extensions/placeholder.js:21]: We don't need `key` here.

File: client/marketing/components/recommended-extensions/style.scss
Comment (ecgan) [reply client/marketing/components/recommended-extensions/style.scss:13]: @tomalec , the [VS Code stylelint extension](https://github.com/stylelint/vscode-stylelint) will show error for you here.

File: client/marketing/components/knowledge-base/test/index.js
Comment (ecgan) [inline client/marketing/components/knowledge-base/test/index.js:55]: @tomalec , I don't see any other files here using `is-preloader` class name, I suppose it comes from other components? (Just want to be sure this is not typo / confusion with `is-placeholder`.

File: client/marketing/components/knowledge-base/placeholder.js
Comment (tomalec) [reply client/marketing/components/knowledge-base/placeholder.js]: Sorry, I forgot to push recent changes 😶 

I was thinking of switching to `div`, but that would require changing CSS, adding more classes, and making it less semantic. AFAIK, for a11y perspective setting `aria-hidden="true"` on any ancestor should do the job - make it hidden for a11y tools. However, for this check, it seems to be not enough. I've added `aria-hidden="true"` explicitly on the `h*` elements to make it happy.

**Comment by ecgan** [comment]
> Personally, I don't like having different files with different classes called the same way. **It makes file searching harder `Ctrl + p`.** In some previous projects, I used to work with conventions to call the file like the class it defines if it defines only one, or at least give it a somewhat-specific module name.

I share the same opinion with you too. An example: when we do `CTRL + P` and search for `index.js`, a lot of files show up, including those from test folders. Also, it makes reading the editor tab title more difficult and takes up more space. Same thing goes with `style.scss` files.

File: client/marketing/components/knowledge-base/placeholder.js
Comment (jconroy) [reply client/marketing/components/knowledge-base/placeholder.js]: Nice, TIL

**Comment by tomalec** [comment]
I created a separate issue regarding file name conventions at https://github.com/woocommerce/woocommerce-admin/issues/5637.

So as @jconroy suggested, I'd stick to `placeholder.js` for now.

File: client/marketing/components/knowledge-base/test/index.js
Comment (tomalec) [reply client/marketing/components/knowledge-base/test/index.js]: Thanks :) It was a typo.

**Comment by jconroy** [comment]
I think the travis build errors are potentially fixed by https://github.com/woocommerce/woocommerce-admin/pull/5635

**Comment by tomalec** [comment]
After merging letest `main`, Travis is all green.
Now, the PR is waiting for another round of reviews :)

**Review by joshuatf — APPROVED** [review]
Very nice addition to the marketing page!  Left a couple quick clean-up comments, but this is testing well and pre-approving.

**Comment by jconroy** [comment]
Thanks for the review @joshuatf

**Comment by timmyc** [comment]
> I definitely see your point though - not sure if @timmyc has any guideance for us.

I don't really have a strong opinion here, and think the p2 that got created around this is a better venue to let folks that do have strong opinions chime in 😆 

I will say that the patterns here in wc-admin largely follow those in Gutenberg, and WordPress.com as well for things like having a duplicated file name.

**Comment by jconroy** [comment]
Thanks for chiming in @timmyc 👍

**Comment by tomalec** [comment]
All comments addressed. I hope I didn't miss anything with translations, as I never used it.
I don't need to update any dictionary files at this point, do I?

**Comment by jconroy** [comment]
> I hope I didn't miss anything with translations, as I never used it. I don't need to update any dictionary files at this point, do I?

@tomalec nope you don't need to do anything with these. We make sure the text is translatable and then translations can be created and shared by others https://translate.wordpress.org/projects/wp-plugins/woocommerce/ there are also a range of plugins available that help with translations.

**Pull request merged**
